src/DotNetty.Codecs/DelimiterBasedFrameDecoder.cs

```diff
@@ -0,0 +1,286 @@
+        ///		NOTE: A see <cref="DotNetty.Codecs.TooLongFrameException" /> is thrown if the length of the frame exceeds this value.</param>
+        /// <param name="stripDelimiter">whether the decoded frame should strip out the delimiter or not</param>
+        /// <param name="failFast">If true, a <cref="DotNetty.Codecs.TooLongFrameException" /> is
+        ///		thrown as soon as the decoder notices the length of the
+        ///		frame will exceed<tt>maxFrameLength</tt> regardless of
+        ///		whether the entire frame has been read.
+        ///		If false, a <cref="DotNetty.Codecs.TooLongFrameException" /> is
+        ///		thrown after the entire frame that exceeds maxFrameLength has been read.</param>
+        /// <param name="delimiter">the delimiter</param>
+        public DelimiterBasedFrameDecoder(int maxFrameLength, bool stripDelimiter, bool failFast, IByteBuffer[] delimiters)
+        {
+            ValidateMaxFrameLength(maxFrameLength);
+            if(delimiters == null)
+                throw new NullReferenceException("delimiters");
+
+            if(delimiters.Length == 0)
+                throw new ArgumentException("empty delimiters");
+
+            if(IsLineBased(delimiters) && !IsSubclass())
+            {
+                lineBasedDecoder = new LineBasedFrameDecoder(maxFrameLength, stripDelimiter, failFast);
+                this.delimiters = null;
+            }
+            else
+            {
+                this.delimiters = new IByteBuffer[delimiters.Length];
+                for(int i = 0; i < delimiters.Length; i++)
+                {
+                    IByteBuffer d = delimiters[i];
+                    ValidateDelimiter(d);
+                    this.delimiters[i] = d.Slice(d.ReaderIndex, d.ReadableBytes);
+                }
+                lineBasedDecoder = null;
+            }
+            this.maxFrameLength = maxFrameLength;
+            this.stripDelimiter = stripDelimiter;
+            this.failFast = failFast;
+        }
+
+        public DelimiterBasedFrameDecoder(int maxFrameLength, IByteBuffer delimiter)
+            : this(maxFrameLength, true, true, new IByteBuffer[] { delimiter })
+        {
+        }
+
+        public DelimiterBasedFrameDecoder(int maxFrameLength, bool stripDelimiter, IByteBuffer delimiter)
+            : this(maxFrameLength, stripDelimiter, true, new IByteBuffer[] { delimiter })
+        {
+        }
+
+        public DelimiterBasedFrameDecoder(int maxFrameLength, bool stripDelimiter, bool failFast, IByteBuffer delimiter)
+            : this(maxFrameLength, true, failFast, new IByteBuffer[] { delimiter })
+        {
+        }
+
+        public DelimiterBasedFrameDecoder(int maxFrameLength, IByteBuffer[] delimiters)
+            : this(maxFrameLength, true, true, delimiters )
+        {
+        }
+
+        public DelimiterBasedFrameDecoder(int maxFrameLength, bool stripDelimiter, IByteBuffer[] delimiters)
```

Comment:
`params IByteBuffer[] delimeters`. Same for other constructors accepting byte buffer array